autorest/azure/async.go

```diff
@@ -434,16 +464,16 @@ func updatePollingState(resp *http.Response, ps *pollingState) error {
 	if ps.hasFailed() {
 		if ps.PollingMethod == PollingAsyncOperation {
 			or := pt.(*operationResource)
-			ps.Code = or.OperationError.Code
-			ps.Message = or.OperationError.Message
+			ps.ServiceError = &or.OperationError
 		} else {
 			p := pt.(*provisioningStatus)
 			if p.hasProvisioningError() {
-				ps.Code = p.ProvisioningError.Code
-				ps.Message = p.ProvisioningError.Message
+				ps.ServiceError = &p.ProvisioningError
```

Comment:
are these type assertions guaranteed to succeed ? 

if so, can we chain the type assertion with the `hasProvisioniningError()` so we reduce the extra if else ?

Reply:
They are guaranteed to succeed as ```pt``` is set to one of the two types at the beginning of the function depending on the type of polling used.  I will clean this up.